### PR TITLE
[Merged by Bors] - Clarify description of API docs

### PR DESCRIPTION
**Pull request by Metadorius**
The way the docs are described on the website may make people think that there are only autogenerated entries in the docs without much useful information for a learner, especially the word "generated" makes such impression. This PR fixes that.

### CODE REVIEW

File: content/learn/book/next-steps/_index.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ You have reached the end of The Bevy Book! And unfortunately, we haven't even sc
 * **[The Bevy Examples](https://github.com/bevyengine/bevy/tree/latest/examples#examples)**: We create an example for every major Bevy feature. This is currently the best way to learn Bevy's features and how to use them.
   * **[The Bevy Web Examples](https://bevyengine.org/examples)**: We also try to make most of these examples available directly in your browser.
 * **[Breakout](https://github.com/bevyengine/bevy/blob/latest/examples/games/breakout.rs)**: A small "breakout" clone that illustrates what a real Bevy game looks like.
-* **[Bevy API Docs](https://docs.rs/bevy)**: Bevy's Rust API documentation.
+* **[Bevy API Docs](https://docs.rs/bevy)**: Bevy's Rust API reference documentation which includes both auto-generated entries and hand-written explanations.
```
Comment (alice-i-cecile) [inline content/learn/book/next-steps/_index.md:15]:
```suggestion
* **[Bevy API Docs](https://docs.rs/bevy)**: Documentation for each type and trait used in Bevy.
```

I don't think that including either "Rust" or "API" is useful here, but it's important to communicate that it's granular.

File: templates/learn.html [hunk 2]
```diff
@@ -24,7 +24,7 @@
         Bevy Rust API Docs
       </div>
       <div class="card-description">
-        Learn about Bevy's Rust API using the generated Rust documentation.
+        Learn Bevy's Rust API in-depth using the reference documentation. Includes both auto-generated entries and hand-written explanations.
```
Comment (alice-i-cecile) [inline templates/learn.html:27]:
```suggestion
        Look up information about Bevy's types, methods and traits using the in-depth reference documentation.
```

Comment (Metadorius) [reply templates/learn.html:27]: Wouldn't such edit(s) still omit that you can also find a usage information on doc pages? For example, I didn't know you could find [a good manual piece on queries](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html) in the docs.

Comment (alice-i-cecile) [reply templates/learn.html:27]:
```suggestion
Learn how to use Bevy's types, methods and traits using the in-depth reference documentation, complete with inline examples.
```
How about that?

Comment (Metadorius) [reply templates/learn.html:27]: Sounds good :)

File: content/learn/book/next-steps/_index.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ You have reached the end of The Bevy Book! And unfortunately, we haven't even sc
 * **[The Bevy Examples](https://github.com/bevyengine/bevy/tree/latest/examples#examples)**: We create an example for every major Bevy feature. This is currently the best way to learn Bevy's features and how to use them.
   * **[The Bevy Web Examples](https://bevyengine.org/examples)**: We also try to make most of these examples available directly in your browser.
 * **[Breakout](https://github.com/bevyengine/bevy/blob/latest/examples/games/breakout.rs)**: A small "breakout" clone that illustrates what a real Bevy game looks like.
-* **[Bevy API Docs](https://docs.rs/bevy)**: Bevy's Rust API documentation.
+* **[Bevy API Docs](https://docs.rs/bevy)**: Documentation and examples for each type, trait and method in Bevy.
```
Comment (Nilirad) [inline content/learn/book/next-steps/_index.md:15]:
```suggestion
* **[Bevy API Docs](https://docs.rs/bevy)**: Documentation with examples for each type, trait and method in Bevy.
```

Examples are part of the documentation.

Comment (Metadorius) [reply content/learn/book/next-steps/_index.md:15]: Somehow the addition of "with" now messes with how the sentence is perceived, like if "for" related to only examplese:
```
(Documentation and examples) for (each type, trait and method) ...
```
vs
``` 
Documentation with (examples for types, traits, methods and other items) ...
```

Comment (Nilirad) [reply content/learn/book/next-steps/_index.md:15]: IMO the ambiguity still remains with “and”.  Also, the reader is able to trivially resolve it.

Comment (alice-i-cecile) [reply content/learn/book/next-steps/_index.md:15]: I like "with" better too :)

**Review by alice-i-cecile — APPROVED** [review]

File: templates/learn.html [hunk 4]
```diff
@@ -24,7 +24,7 @@
         Bevy Rust API Docs
       </div>
       <div class="card-description">
-        Learn about Bevy's Rust API using the generated Rust documentation.
+Learn how to use Bevy's types, traits and methods using the in-depth reference documentation, complete with inline examples.
```
Comment (mockersf) [inline templates/learn.html:27]: could you indent this line?

Comment (Metadorius) [reply templates/learn.html:27]: Not sure why it suddenly gone unindented, will do.

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**